library/L1_Peripheral/lpc17xx/system_controller.hpp

```diff
@@ -117,273 +53,563 @@ class SystemController final : public sjsu::SystemController
     static constexpr auto kGpdma             = PeripheralID::Define<29>();
     static constexpr auto kEthernet          = PeripheralID::Define<30>();
     static constexpr auto kUsb               = PeripheralID::Define<31>();
-    static constexpr auto kCpu               = PeripheralID::Define<32>();
     //! @endcond
   };
 
-  /// Namespace of Oscillator register bitmasks
-  struct Oscillator  // NOLINT
+  /// LPC17xx Peripheral Clock Selection Values:
+  /// The kDeviceId of each peripheral clock corresponds to the peripheral's
+  /// clock selection bit position within the LPC17xx System Controller's
+  /// PCLKSEL registers.
+  class Clocks
```

Comment:
I came to realize that my previous implementation of setting the peripheral divider was incorrect since now all power control bits in the PCONP register correspond to the peripheral divider select bits in the PCLK_SEL0 and PCLK_SEL1 registers.